scripts/Muon/GUI/ElementalAnalysis2/context/context.py

```diff
@@ -83,16 +109,24 @@ def handle_calculation_error(self, error):
     def _run_rebin(self, name, rebin_type, params):
         rebined_run_name = None
         if rebin_type == "Fixed":
+            rebin_index = 1
+            rebin_option = "Steps: " + str(params) + " KeV"
             rebined_run_name = str(name) + REBINNED_FIXED_WS_SUFFIX
         if rebin_type == "Variable":
+            rebin_index = 2
+            rebin_option = "Bin Boundaries: " + str(params)
             rebined_run_name = str(name) + REBINNED_VARIABLE_WS_SUFFIX
 
         remove_ws_if_present(rebined_run_name)
 
-        workspace = Rebin(InputWorkspace=name, OutputWorkspace=rebined_run_name, Params=params)
-        group = retrieve_ws(name.split(";")[0])
-        group.addWorkspace(workspace)
-        self.group_context[name].update_workspaces(name, workspace, rebin=True)
+        workspace = Rebin(InputWorkspace=self.group_context[name].get_counts_workspace_for_run(),
+                          OutputWorkspace=rebined_run_name, Params=params)
+
+        group_workspace = retrieve_ws(self.group_context[name].run_number)
```

Comment:
you should use this for ADS calls https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/ADSHandler/ADS_calls.py